src/widgets/toggle-refinement/__tests__/toggle-refinement-test.ts

```diff
@@ -38,499 +38,347 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/toggle-refi
-            attribute,
-            true
-          );
-          expect(helper.addDisjunctiveFacetRefinement).not.toHaveBeenCalled();
-        });
-      });
-      describe('specific values', () => {
-        it('toggle on should change the refined value', () => {
-          // Given
-          const userValues = { on: 'on', off: 'off' };
-          widget = toggleRefinement({
-            container: containerNode,
-            attribute,
-            ...userValues,
-          });
-
-          const config = widget.getWidgetSearchParameters(
-            new SearchParameters({}),
-            { uiState: {} }
-          );
-          const altHelper = jsHelper(createSearchClient(), '', config);
-
-          const createURL = () => '#';
-
-          widget.init!(createInitOptions({ helper: altHelper, createURL }));
-
-          // When
-          toggleOn({ createURL, altHelper });
-
-          // Then
-          expect(
-            altHelper.state.isDisjunctiveFacetRefined(attribute, 'off')
-          ).toBe(false);
-          expect(
-            altHelper.state.isDisjunctiveFacetRefined(attribute, 'on')
-          ).toBe(true);
-        });
-
-        it('toggle off should change the refined value', () => {
-          // Given
-          const userValues = { on: 'on', off: 'off' };
-          widget = toggleRefinement({
-            container: containerNode,
-            attribute,
-            ...userValues,
-          });
-          widget.getWidgetSearchParameters(new SearchParameters({}), {
-            uiState: {},
-          });
-          helper.state.isDisjunctiveFacetRefined = jest
-            .fn()
-            .mockReturnValue(true);
-          const createURL = () => '#';
-          widget.init!(createInitOptions({ helper, createURL }));
-
-          // When
-          toggleOff({ createURL });
-
-          // Then
-          expect(helper.removeDisjunctiveFacetRefinement).toHaveBeenCalledWith(
-            attribute,
-            'on'
-          );
-          expect(helper.addDisjunctiveFacetRefinement).toHaveBeenCalledWith(
-            attribute,
-            'off'
-          );
+        const search = instantsearch({
+          indexName: 'test',
+          initialUiState: {
+            test: {
+              toggle: {
+                [attribute]: true,
+              },
+            },
+          },
+          searchClient: createSearchClient({
+            search(queries) {
+              return Promise.resolve({
+                results: queries.map(() =>
+                  createSingleSearchResponse({
+                    facets: {
+                      [attribute]: {
+                        true: 2,
+                        false: 1,
+                      },
+                    },
+                  })
+                ),
+              });
+            },
+          }),
         });
-      });
-    });
-
-    describe('custom off value', () => {
```

Comment:
these tests still exist in the connector, but don't need to exist in the widget, IMO